File: project-dr-web/src/app/dr/public-notice-web-crawler/entity/public-notice-web-crawler-search-form.ts
import {Page} from '@sb/base';


export interface PublicNoticeWebCrawlerSearchForm extends Page {



    /**
     * 发布网站
     */
    publicWeb?: string;

    /**
     * 标题
     */
    title?: string;


    /**
     * 发布时间
     */
    publicTime?: string;

    /**
     * 发布开始时间
     */
    publicTimeStart?: string;
    /**
     * 发布结束时间
     */
    publicTimeEnd?: string;

    /**
     * 省份
     */
    province?: string;

    /**
     * 招标方式
     */
    bidType: string;
    /**
     * 采购人
     */
    purchaseUser: string;
    /**
     * 采购代理机构
     */
    purchaseOrganization: string;
    /**
     * 是否已经解析
     */
    ifAnalysis: string;

}

File: project-dr/src/main/java/com/sbwork/dr/publicnoticewebcrawler/searchForm/PublicNoticeWebCrawlerSearchForm.java
package com.sbwork.dr.publicnoticewebcrawler.searchForm;

import com.sbwork.base.persistence.annotate.Query;
import com.sbwork.base.persistence.query.bean.QueryType;
import com.sbwork.systemConfig.plugin.bean.Page;
import io.swagger.annotations.ApiModelProperty;
import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;
import java.util.Date;

/**
 * 公告资源爬取搜索类
 */
@Getter
@Setter
public class PublicNoticeWebCrawlerSearchForm extends Page implements Serializable {

    /**
     * 发布开始时间
     */
    @ApiModelProperty(value = "发布开始时间")
    @Query(value = QueryType.gte,fieldName = "publicTime")
    private Date publicTimeStart;

    /**
     * 发布结束时间
     */
    @ApiModelProperty(value = "发布结束时间")
    @Query(value = QueryType.lte,fieldName = "publicTime")
    private Date publicTimeEnd;

    @ApiModelProperty(value = "发布网站")
    @Query(value = QueryType.like)
    private String publicWeb;

    @ApiModelProperty(value = "标题")
    @Query(value = QueryType.like)
    private String title;

    @ApiModelProperty(value = "省份")
    @Query(value = QueryType.like)
    private String province;

    @ApiModelProperty(value = "招标方式")
    @Query(value = QueryType.like)
    private String bidType;

    @ApiModelProperty(value = "是否已经解析")
    @Query(value = QueryType.equal)
    private String ifAnalysis;
}
